src/cmd/ksh93/tests/pty.sh

```diff
@@ -826,5 +826,22 @@ w :\E_
 r ^:test-2: : One\\ "Two Three"\$'Four Five'\.mp3\r\n$
 !
 
+# err_exit #
+((SHOPT_VSH)) && tst $LINENO <<"!"
+L crash when entering comment into history file (vi mode)
+# https://github.com/att/ast/issues/798
+
+d 15
+p :test-1:
+w set -o vi
+r ^:test-1: set -o vi\r\n$
```

Comment:
The `set -o vi` is not needed, it's the default as `VISUAL` is set to `vi`.

Also, for testing against emacs bugs, I tend to use `VISUAL=emacs tst $LINENO <<"!"` so that the test doesn't need to issue a `set` command.